### ISSUE TITLE
Inconsistencies with Style/ZeroLengthPredicate

### ISSUE THREAD

**Issue by herwinw** [rubocop-hq/rubocop#7089]
Proof of concept:

```ruby
def foo(arr)
  return true if arr.empty?
  return true if arr.size.zero?
  return true if arr.size == 0
  return true if arr.length.zero?
  return true if arr.length == 0
  return true if arr.count.zero?
  return true if arr.count == 0

  false
end
```

--------

## Expected behavior

I would expect an offense for `Style/ZeroLengthPredicate` for every line that starts with `return true if`except for the frist one.

## Actual behavior

There are two problems here:

1. The method `Array#count` does not get detected in `Style/ZeroLengthPredicate`, most likely because it can accept an extra argument or block to filter, but we don't use that here. I would expect the lines with `arr.count` to have an update to `empty?` as well.
2. The line `arr.size == 0` triggers both `Style/NumericPredicate` (to convert it to `arr.size.zero?`) and `Style/ZeroLengthPredicate` (to convert it to `arr.empty?`). The statement `arr.size.zero?` is accepted by Rubocop, but I would expect this to be corrected to `arr.empty?` as well. `length` is handled the same as `size`, `count` is skipped by `Style/ZeroLengthPredicate` (see item 1)/

## Steps to reproduce the problem

A PoC has been added. It triggers a few additional warnings as well (two complexity metrics and a missing frozing string literals), but those are unrelated to the bug report and can be ignored.

## RuboCop version

```
0.70.0 (using Parser 2.6.3.0, running on ruby 2.3.3 i386-linux-gnu)
```
